packages/@aws-cdk/core/lib/resource.ts

```diff
@@ -168,3 +214,35 @@ export abstract class Resource extends Construct implements IResource {
     });
   }
 }
+
+class EnvComponent implements IEnvComponent {
```

Comment:
Is it valuable to have this be a private implementation of a public interface type? For simple value types there seems to be no value in subclassing, so why not just have `EnvComponent` be a public class?

Reply:
Are there any downsides to this that you see?

Reply:
Changed.